compiler/Makefile

```diff
@@ -123,6 +123,19 @@ TARGETS = $(CHPL)
 
 LIBS = -lm
 
+ifeq ($(CHPL_MAKE_HOST_MEM), jemalloc)
+ifeq ($(CHPL_MAKE_PLATFORM), darwin)
+JEMALLOC_LIBS=-L$(shell jemalloc-config --libdir) -ljemalloc $(shell jemalloc-config --libs)
+else
+ifeq ($(CHPL_LIB_PIC), jemalloc)
+JEMALLOC_LIBS=$(JEMALLOC_HOST_LIB_DIR)/libjemalloc_pic.a
+else
+JEMALLOC_LIBS=$(JEMALLOC_HOST_LIB_DIR)/libjemalloc.a
```

Comment:
What's the motivation for explicitly specifying the `.a` instead of sticking with `-L`/`-l`. We only build the .a for our bundled version anyways, so there's no shared lib to take precedence. Oh, maybe this is concern about finding a system version instead of finding our bundled version?

I'd personally be in favor of `-L`/`-l` and we probably need to use `jemalloc-config` still to get any extra link args (like say `-lm  -lpthread`)

Reply:
(And it might be nice to put that logic into a chplenv script to mimic how we're getting links args for the target build)

Reply:
Due to the lack of `-lpthread` this fails for me on raptor with:

```
/lus/cls04005/eronagha/chapel/third-party/jemalloc/jemalloc-src/src/mutex.c:103: undefined reference to `pthread_mutexattr_destroy'
/lus/cls04005/eronagha/chapel/third-party/jemalloc/install/host/cray-cs-x86_64-none-gnu-none/lib/libjemalloc.a(tcache.o): In function `chpl_je_tsd_set':
```

Resolving the undefined reference should be easy, though I thought it was weird the function name was prefixed with `chpl_je` still.

Reply:
Oh, nevermind, that's from `--with-private-namespace=chpl_`, not because we're still specifying a `--with-jemalloc-prefix`

Reply:
Updated to use `jemalloc-config` for bundled builds in `chpl_jemalloc.py`. We should do the same for `system` right?

Reply:
System is only supported with dynamic libraries today, right? So shouldn't matter since the dynamic libs will figure out their dependencies.